scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -218,16 +227,22 @@ def handle(self, line):
                 self.recording.append(csv)
 
         self.process_test(line)
-        # Reset the resulting test state to 'failed' for 'one_line' and
-        # ordered 'multi_line' patterns when not all of these patterns were
+        # Reset the resulting test state to 'failed' when not all of the patterns were
         # found in the output, but just ztest's 'PROJECT EXECUTION SUCCESSFUL'.
         # It might happen because of the pattern sequence diverged from the
         # test code, the test platform has console issues, or even some other
         # test image was executed.
-        # TODO: Introduce explicit match policy type either to reject
-        # unexpected console output, or to allow missing patterns.
+        # TODO: Introduce explicit match policy type to reject
+        # unexpected console output, allow missing patterns, deny duplicates.
         if self.state == "passed" and self.ordered and self.next_pattern < self.patterns_expected:
-            logger.error(f"HARNESS:{self.__class__.__name__}: failed with only {self.next_pattern} matched patterns from expected {self.patterns_expected}")
+            logger.error(f"HARNESS:{self.__class__.__name__}: failed with"
+                         f" {self.next_pattern} of {self.patterns_expected}"
```

Comment:
`{self.next_pattern+1}`

Reply:
> `{self.next_pattern+1}`

why ? 
for `debug.coredump.backends.logging` it logs `ERROR   - HARNESS:Console: failed with 2 of 4 expected ordered patterns. `
for:
```
regex:                                                                                                                                                                                                                                       
  - "E: #CD:BEGIN#"                                                                                                                                                                                                                            
  - "E: #CD:5([aA])45([0-9a-fA-F]+)"                                                                                                                                                                                                          
  - "FOOO"
  - "E: #CD:END#"
```

Reply:
Ok, I see it now. It is ordered pattern, so it receives number of matched patterns before failing one. First I thought that should be `3 of 4` in that case.